File: frontend/src/types/room.ts
export type RoomType = 'GUEST_HOUSE' | 'FRAME';

export interface Room {
  id: number;
  number: string;
  type: RoomType;
  floor: number;
  capacity: number;
  price_per_night: number;
  is_available: boolean;
  description?: string;
  amenities?: string;
  created_at: string;
  updated_at: string;
}

export interface RoomCreate {
  number: string;
  type: RoomType;
  floor: number;
  capacity: number;
  price_per_night: number;
  description?: string;
  amenities?: string;
}

export interface RoomUpdate extends Partial<RoomCreate> {
  is_available?: boolean;
} 

File: backend/app/schemas/room.py
from pydantic import BaseModel
from typing import Optional
from app.models.hotel import RoomType

class RoomBase(BaseModel):
    number: str
    type: RoomType
    floor: int
    capacity: int
    price_per_night: float
    description: Optional[str] = None
    amenities: Optional[str] = None

class RoomCreate(RoomBase):
    pass

class RoomUpdate(BaseModel):
    number: Optional[str] = None
    type: Optional[RoomType] = None
    floor: Optional[int] = None
    capacity: Optional[int] = None
    price_per_night: Optional[float] = None
    is_available: Optional[bool] = None
    description: Optional[str] = None
    amenities: Optional[str] = None

class Room(RoomBase):
    id: int
    is_available: bool

    class Config:
        from_attributes = True 